template_parser.py

```diff
@@ -97,19 +97,27 @@ def read_template_types(template_type, container_cmd, path_dirs):
     delay_sections = {}
 
     try:
+        # list of containers enabled by modes.template
         mode_enabled = ast.literal_eval(subprocess.check_output("python2.7 "+info_dir+"get_status.py menabled -b "+path_dirs.base_dir, shell=True))
-        if template_type in mode_enabled:
-            mode_enabled = mode_enabled[template_type]
-        elif template_type in ['active', 'passive'] and 'collectors' in mode_enabled:
-            mode_enabled = [container for container in mode_enabled['collectors'] if container.startswith(template_type+"-")]
-
-        # filters out core_disabled containers from modes_enabled list.
+        # filter out core_disabled containers from modes_enabled. For collectors, containers must be in mode_enabled and core_enabled to be enabled
         core_enabled, core_disabled = ast.literal_eval(subprocess.check_output("python2.7 "+info_dir+"get_status.py cenabled -b "+path_dirs.base_dir, shell=True))
-        if template_type in core_disabled:
-            core_disabled = core_disabled[template_type]
-        elif template_type in ['active', 'passive'] and 'collectors' in core_disabled:
-            core_disabled = core_disabled['collectors']
-        sections = [container for container in mode_enabled if not container in core_disabled]
+
+        if template_type in ['active', 'passive']:
+            if 'collectors' in mode_enabled and 'collectors' in core_enabled:
+                mode_enabled = [container for container in mode_enabled['collectors'] if container.startswith(template_type+"-")]
+                core_enabled = core_enabled['collectors']
+                sections = [container for container in mode_enabled if container in core_enabled]
+            else:
+                sections = []
```

Comment:
why not declare this from the beginning, say `line 104` then you don't have to do these additional checks.